swarm/storage/netstore.go

```diff
@@ -221,14 +219,16 @@ func (n *NetStore) RemoteFetch(ctx context.Context, req *Request, fi *FetcherIte
 		osp.LogFields(olog.String("ref", ref.String()))
 
 		log.Trace("remote.fetch", "ref", ref)
-		currentPeer, err := n.RemoteGet(innerCtx, req, n.localID)
+
+		nctx := context.WithValue(innerCtx, "remote.fetchh", osp)
```

Comment:
fetchh -> fetch

Reply:
This is used to propagate the `osp` span to `FindPeer`. Let's make `FindPeer` actually work, before we make the code nice. Remember `ctx` is for optional values that must have no effect on the business logic of the code.